include/boost/math/statistics/bivariate_statistics.hpp

```diff
@@ -8,25 +9,28 @@
 
 #include <iterator>
 #include <tuple>
+#include <type_traits>
+#include <cmath>
+#include <cstddef>
 #include <boost/assert.hpp>
 
 
-namespace boost{ namespace math{ namespace statistics {
+namespace boost{ namespace math{ namespace statistics { namespace detail {
 
-template<class Container>
-auto means_and_covariance(Container const & u, Container const & v)
+template<class ReturnType, class Container>
```

Comment:
This will be a breaking change.

Reply:
It is a breaking change so it got renamed and moved into the detail namespace. The user facing functions remain the same and offer return type deduction via `std::enable_if` (lines 93-103). I can move implementation details into a separate file to make this more clear. I did not do that already because this file is still pretty small.